packages/react-composites/tests/browser/chat/app/fake-back-end/FakeChatClient.ts

```diff
@@ -85,6 +85,11 @@ export class FakeChatClient implements IChatClient {
     });
   }
 
+  initializeMessages(threadId: string, messages: BasicChatMessage[]): Promise<void> {
```

Comment:
Let's keep the fake completely isolated from test logic.

The constructor for `FakeChatClient` already accepts a preconstructed `Model` to initialize with. The test should construct the appropriate model and initialize the `FakeChatClient` with the `Model`.

Reply:
UPDATE:

I see that `Model` does not have to initialize the thread and messages.

You have two options:

1. Add a constructor for `Model` that lets you initialize the data. But use the data types from the headless SDK. The fake uses the headless DTOs. No need to invent another one.
2. Simply use the `ChatClient` that you have created to create a thread, add users and send messages as if you're using the real service.

I recommend (2). It needs not new API in the fake, and is easier to extend (you can get the fake in any state you want by using the public headless API, at least until we want error injection).